### PR TITLE
🔧 chore(carousel): Update zIndex to use fallback value

### LINKED ISSUE #1101: If a theme is applied, use the theme's token in the `Caroucel`, `CarouselIndicators` and `CarouselControl` component
### Which package has an issue?

@yamada-ui/react

### Package Version

1.3.0

### Description

Currently, styling values in components ignore theme tokens. If a theme is applied, you should use the theme's token.

Yamada UI's style system provides fallback functions. If the value of the token in the first argument does not exist, it returns the second argument, the fallback value.

```diff
const css: CSSUIObject = {
  position: "absolute",
-  zIndex: 9,
+  zIndex: "fallback(kurillin, 9)",
  display: "flex",
  justifyContent: "center",
  ...styles.indicators,
  ...(orientation === "vertical"
    ? { flexDirection: "column" }
    : { flexDirection: "row" }),
}
```

```diff
const css: CSSUIObject = {
  position: "absolute",
-  zIndex: 9,
+  zIndex: "fallback(kurillin, 9)",
  ...styles.control,
  ...styles[operation],
}
```

```diff
/**
 * The CSS `gap` property.
 *
 * @default '4'
 */
gap?: CSSUIProps["gap"]
```

```diff
export const useCarousel = ({
  index,
  defaultIndex = 0,
  onChange,
  align = "center",
  orientation = "horizontal",
  autoplay = false,
  stopMouseEnterAutoplay = true,
  loop = true,
  speed = 10,
  delay = 4000,
-  gap = "1rem",
+  gap = "fallback(4, 1rem)",
  slidesToScroll = 1,
  draggable = true,
  dragFree = false,
  inViewThreshold = 0,
  skipSnaps = false,
  containScroll = "",
  slideSize = "100%",
  includeGapInSize = true,
  onScrollProgress,
  children,
  ...rest
}: UseCarouselProps) => {
```

```diff
const css: CSSUIObject = {
  display: "flex",
  flexDirection: orientation === "vertical" ? "column" : "row",
  ...styles.inner,
  ...(includeGapInSize
-    ? { [orientation === "vertical" ? "mb" : "mr"]: `-${gap}` }
+    ? {
+        var: [{ __prefix: "ui", name: "gap", token: "spaces", value: gap }],
+        [orientation === "vertical" ? "mb" : "mr"]:
+          calc.negate("var(--ui-gap)"),
      }
    : {}),

}
```

### Link to Reproduction

_No response_

### Steps to Reproduce

2. Click on '...'
3. Scroll down to '...'

### Browser

None

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

### Do you know how to fix the issue?

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by KekmaTime**
Closes #1101

## Description
- Added support for utilizing theme tokens in the Carousel, CarouselIndicators, and CarouselControl components.

## Current behavior (updates)

- Styling values in components ignore theme tokens.
- Theme tokens are not utilized for styling which leads to inconsistency when themes are applied.

## New behavior

- Updated the components to utilize theme tokens for styling values.

## Is this a breaking change (Yes/No):

- Not sure yet

## Additional Information

I've made the CSS  calc function inside backticks so that it can be used in the JS/TS context

### CODE REVIEW

**Review by komura-c — CHANGES_REQUESTED** [review]
@KekmaTime 
Sorry for the late review. Please run `pnpm changeset` and select `patch`.

**Comment by KekmaTime** [comment]
> Sorry for the late review. Please run `pnpm changeset` and select `patch`.
can you tell how what patch means? im new to using changeset 😅

**Comment by komura-c** [comment]
@KekmaTime 
The patch means a version with backward compatible bug fixes in semantic versioning.
This will be used when the package is released and will be noted in related package's [`CHANGELOG.md`.](https://github.com/yamada-ui/yamada-ui/blob/main/packages/components/carousel/CHANGELOG.md#patch-changes-6)

**Comment by hirotomoyamada** [comment]
@KekmaTime 

Hi, thanks for the PR.

In this PR, what you fixed is `@yamada-ui/carousel`. It looks like you are trying to upgrade the version of `@yamada-ui/react`. Also, this fix is probably a patch, not a major one.

So,

```
"@yamada-ui/react": major
```

to

```
"@yamada-ui/carousel": patch
```

File: .changeset/dull-pumpkins-love.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/react": major
```
Comment (hirotomoyamada) [inline .changeset/dull-pumpkins-love.md:2]: Change `"@yamada-ui/carousel": patch`

**Review by hirotomoyamada — APPROVED** [review]

**Review by komura-c — APPROVED** [review]
Approved

**Pull request merged**